frame/ethereum/src/lib.rs

```diff
@@ -360,50 +367,99 @@ impl<T: Trait> Module<T> {
 					transaction.gas_price,
 					Some(transaction.nonce),
 					true,
-				).unwrap(); // TODO: handle error
+				);
+
+				match execution {
```

Comment:
@sorpaas I don't find this code very elegant.
Is there a better way to handle the error in those cases ?

Reply:
Why not just use `try!` macro (`?` syntax).

By the way, I don't think even returning errors in `ExitReason::Fatal` is entirely valid. We still write gas changes to storage even in that case.

Reply:
Thanks, I'll try that :)
Also, for the `transact` function. My understanding is that it must return `..,,DispatchError>` but those error can't contain data. Is there an easy way to pass the reason of the error with it, or I have to enum all the possible errors ? 
(Ex: The EVM might return with ExitRevert and the message be `out of gas`. Should I add `OutOfGas` in the `decl_error! Error` ?)

Reply:
As far as I know the `DispatchError`s are basically strings. If it returns error, I think you can safely just pass it upwards. All the EVM internal errors, like out-of-gas, cannot be returned via `DispatchError`, because in those situations we still change state. We can return events signaling those errors.

However, if you really want to be precise, we can define module functions that return proper errors for those EVM functions. Just make sure when using them in dispatchables those errors are converted to `Ok(())` with error events.

Reply:
Thank you @sorpaas.
So if we have to convert it to an Ok, how can we make it return the error "Out of Gas" from the RPC response directly (I know the event is a good, and standard solution for substrate, but the Eth RPC seems to return the Out of gas directly when replying to the sendRawTransaction)